File: Python/Level2/영어 끝말잇기.py
def solution(n, words):
    word_set = set()
    for i in range(len(words)):
        # 전 단어와 끝말잇기가 되지 않은 경우
        if i > 0 and words[i][0] != words[i - 1][-1]:
            return [i % n + 1, i // n + 1]
        # 중복된 단어를 제시한 경우
        elif words[i] in word_set:
            return [i % n + 1, i // n + 1]
        # 단어를 알맞게 제시한 경우
        else:
            word_set.add(words[i])
    return [0, 0]

File: programmers-coding-test/C++/Level2/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> solution(int n, vector<string> words) {

	int len = words.size();
	unordered_map<string, int> hmap;

	for (int i = 0; i < len; i++)
	{
		if (i > 0 && words.at(i).at(0) != words.at(i - 1).at(words.at(i - 1).length() - 1)) return { i % n + 1, i / n + 1 };
		else if (hmap.count(words.at(i))) return { i % n + 1, i / n + 1 };
		else hmap[words.at(i)]++;
	}

	return vector<int>(2, 0);
}
